akka-stream/src/main/scala/akka/stream/Queue.scala

```diff
@@ -35,3 +43,29 @@ trait SinkQueue[T] {
    */
   def pull(): Future[Option[T]]
 }
+
+sealed abstract class StreamCallbackStatus[+T]
+
+/**
+ * Contains types that is used as return types for async callbacks to streams
+ */
+object StreamCallbackStatus {
```

Comment:
I am not sure this helps in this case given that you are unlikely reuse enqueueing logic as dequeueing logic and vice versa, which means that there is no real use case to make this generic (apart from having one less class, but I think more meaningful, domain specific return types are worth the effort here)